### ISSUE TITLE
Can't install xmonad and xmobar due to GHC errors

### ISSUE THREAD

**Issue by robberer** [NixOS/nixpkgs#5980]
```
Configuring xmonad-0.11...
Flags chosen: testing=False, small_base=True
Dependency X11 >=1.5 && <1.7: using X11-1.6.1.2
Dependency base >=3 && <5: using base-4.7.0.2
Dependency containers -any: using containers-0.5.5.1
Dependency directory -any: using directory-1.2.1.0
Dependency extensible-exceptions -any: using extensible-exceptions-0.1.1.4
Dependency filepath -any: using filepath-1.3.0.2
Dependency mtl -any: using mtl-2.1.3.1
Dependency process -any: using process-1.2.0.0
Dependency unix -any: using unix-2.7.0.1
Dependency utf8-string ==0.3.*: using utf8-string-0.3.8
Setup: The following installed packages are broken because other packages they
depend on are missing. These broken packages must be rebuilt before they can
be used.
package utf8-string-0.3.8 is broken due to missing package
bytestring-0.10.4.0-d6f1d17d717e8652498cab8269a0acd5
./gnulib/tests/test-mbrtowc1.sh: interpreter directive changed from "/bin/sh" to "/nix/store/cxrc12hkmvrjlq747cm08av6jl41nzi1-bash-4.3-p30/bin/sh  "
./gnulib/tests/test-vc-list-files-git.sh: interpreter directive changed from "/bin/sh" to "/nix/store/cxrc12hkmvrjlq747cm08av6jl41nzi1-bash-4.3-p30/bin/sh  "
./gnulib/tests/test-perror.sh: interpreter directive changed from "/bin/sh" to "/nix/store/cxrc12hkmvrjlq747cm08av6jl41nzi1-bash-4.3-p30/bin/sh  "
builder for ‘/nix/store/q5f0zda54m7cl2hph9jajxv25389ap70-haskell-xmonad-ghc7.8.4-0.11-shared.drv’ failed with exit code 1
error (ignored): cannot unlink ‘/nix/store/lsf0dj0pwp33yqkpahswdqjszi4gqfhc-X11-fonts.drv.chroot/tmp/nix-build-X11-fonts.drv-0’: Device or resource busy
error (ignored): cannot unlink ‘/nix/store/8qs1mfnhwqsnk5v3h7xb1vvjzhdsldhj-haskell-X11-xft-ghc7.8.4-0.3.1-shared.drv.chroot/tmp/nix-build-haskell-X11-xft-ghc7.8.4-0.3.1-shared.drv-0’: Device or resource busy
cannot build derivation ‘/nix/store/79c0anq8pfwp0h77lkiyimndc9fjbii6-haskell-env-ghc-7.8.4.drv’: 1 dependencies couldn't be built
```

I'm not using Haskell at all so I'm stuck on this error. No idea how to solve it.

**Comment by peti** [comment]
You have to garbage-collect all Haskell related packages on your system and re-download them from scratch from Hydra to make sure that all the binaries fit together.

**Comment by copumpkin** [comment]
Seems weird to require a GC in a nix world. Is that just a precaution? If you have a pure nix-shell that doesn't refer to the old packages, the new Haskell/Xmonad/Xmobar world should not know anything about them, right?

**Comment by peti** [comment]
GHC produces non-deterministic library ids. The issue above occurred because a locally built library was assign an ID that differs from the one generated on Hydry, and now packages that refer to the Hydra-built library cannot link the one that was compiled locally even though the two builds are, for all practical purposes, identical and have the same Nix store path. It sucks, but there's nothing we can do about it in Nix; this has to be fixed in GHC (and apparently 7.10.1 fixes or at least alleviates the problem).
